solar_system.rb

```diff
@@ -0,0 +1,211 @@
+
+#SETTING CLASSES
+class Planet
+  attr_reader :name, :year_length, :distance_from_sun, :type, :moons,
+  :inhabitants
+
+  def initialize(name, year_length, distance_from_sun, type, moons, inhabitants)
+    @name = name
+    @year_length = year_length
+    @distance_from_sun = distance_from_sun
+    @type = type
+    @moons = moons
+    @inhabitants = inhabitants
+  end
+
+  def planet_all_info
+    output = ""
+    output << "\n    Planet name: #{name}
+    Length of year in earth-days: #{year_length}
+    Distance from the sun in some units: #{distance_from_sun}
+    Class type: #{type}
+    Number of Moons: #{moons}
+    Home to: #{inhabitants}\n "
+    return output
+  end
+end
+
+class SolarSystem
+  attr_accessor :planets
+
+  def initialize(planets)
+    @planets = planets
+  end
+
+  def add_planet(new_planet)
+    @planets << new_planet
+  end
+
+  def ss_name_list
+    i=1
+    name_list = ""
+    planets.each do |planets|
+      name_list<< "#{i}. #{planets.name}\n"
+      i +=1
+    end
+    return name_list
+  end
+
+end
+
+#DEFINING METHODS
+#is the user input a y or a n?  if not, keep cylcing through
+def y_or_n(input)
+  until input == "Y" || input == "N"
```

Comment:
I really like that you've taken all these bits of code for handling the user and turned them into independent methods - good instinct! It makes the driver code at the bottom very concise and readable.